horovod/common/hashes.h

```diff
@@ -33,6 +34,26 @@ template <typename T> struct hash<std::vector<T>> {
   }
 };
 
+template <typename U, typename V> struct hash<std::tuple<U, V>> {
+  typedef std::tuple<U, V> argument_type;
+  typedef std::size_t result_type;
+
+  result_type operator()(argument_type const& in) const {
+    size_t seed = 0;
+    seed ^= std::hash<U>()(std::get<0>(in)) + 0x9e3779b9 + (seed << 6) +
```

Comment:
why this 0x9e3779b9 specific value? can we define it as a constant and not use the magic number?